trotters/trotters.py

```diff
@@ -0,0 +1,259 @@
+import logging
+import hashlib
+import asyncio
+import discord
+from datetime import datetime
+import time
+
+from redbot.core import checks, commands, Config, modlog
+from redbot.core.bot import Red
+from redbot.core.config import Group
+from redbot.core.commands import Context, Cog
+from redbot.core.utils.predicates import MessagePredicate
+
+try:
+    from redbot.core.commands import GuildContext
+except ImportError:
+    from redbot.core.commands import Context as GuildContext
+
+
+class Trotters(commands.Cog):
+    __author__ = "TheBluekr#2702"
+
+    def __init__(self, bot):
+        super().__init__()
+        self.bot = bot
+        self.logger = logging.getLogger("aurelia.cogs.trotters")
+        self.config = Config.get_conf(self, identifier=428038701, force_registration=True)
+
+        default_role = {"addable": False, "USERS": {}}
+
+        self.config.register_role(**default_role)
+
+    def cog_unload(self):
+        pass
+
+    async def initialize(self):
+        await self.register_casetypes()
+
+    @staticmethod
+    async def register_casetypes():
+        # register mod case
+        role_case = {
+            "name": "Role List Update",
+            "default_setting": True,
+            "image": "\N{PAGE FACING UP}",
+            "case_str": "roleupdate",
+        }
+        try:
+            await modlog.register_casetype(**role_case)
+        except RuntimeError:
+            pass
+
+    # Commands
+    @commands.group()
+    @commands.guild_only()
+    async def trotters(self, ctx: Context):
+        """Trotter's Pass settings"""
+        pass
+
+    @checks.admin_or_permissions(manage_roles=True)
+    @trotters.command(name="set")
+    async def set_role(self, ctx, role: discord.Role, enabled: bool):
```

Comment:
make sure to set the types for context everywhere. you really _dont_  need to set it for context since that's always a given, but if you are make sure you are consistent.